### ISSUE TITLE
Crawl suddenly stops for no apparent reason

### ISSUE THREAD

**Issue by tedbjorling** [cgiffard/node-simplecrawler#192]
When I run the following code on 0.5.4 the crawler suddenly stops for no obvious reason (in the example below, I usually see that it stops when having reached approx. ~14800 queue items):

```javascript
var Crawler = require("simplecrawler");
var c = new Crawler("www.ahlens.se")
c.on("fetchcomplete", function(queueItem) {
	console.log("%d items in queue - %s",this.queue.getWithStatus('queued').length, queueItem.url);
});
c.addFetchCondition(function(obj) {
	if ((obj.path.match(/\.css/i) != null)
	 || (obj.path.match(/\.ico/i) != null)
	 || (obj.path.match(/\.jpg$/i) != null)
	 || (obj.path.match(/\.gif$/i) != null)
	 || (obj.path.match(/\.png$/i) != null)
	) {
		return false;
	}
	return true;
});
c.start();
```

If I freeze/defrost the queue and resume the crawl session, it continues for essentially the same duration after which it stops again, and so on.

**Comment by cgiffard** [comment]
So it fires `fetchcomplete`, even where the queue has a considerable number of unfetched items? :-/

**Comment by tedbjorling** [comment]
I guess the `fetchcomplete` events are expected when resources have been fetched and are ready for processing. The `complete` is never emitted, so basically it just seems to be left hanging there. Are you able to reproduce this?

**Comment by nttdocomo** [comment]
I have the same problem

**Comment by tedbjorling** [comment]
After having done some debuging I made two fixes that appears to have remedied this for me at least. I realize I could have been a bit more methodical, but fwiw, the following greased the gears for me:

1) I was using an older version of Node.js (0.10.33), upgrading to 5.1.1 made things run much smoother.

2) I saw issues with the "receiveData" function in crawler.js, where it sometimes did not enter the code block calling processReceivedData(). Reason being that the http(s) response did return an amount of data less than the states "content-length", meaning that things essentially came to an halt. If I send through wether the http(s) response is "data" or "end" to  the "receiveData" function, and ensure that the code block calling processReceivedData() is invoked if the http(s) response has come to an end, I experience no more issues of the kind stated above.

**Comment by fredrikekelund** [comment]
@tedbjorling could you submit your changes in a PR? Sounds like this could do some good if merged

**Comment by fredrikekelund** [comment]
I've tried to address this in #209, would be great with some additional testing of that fork! Ping @tedbjorling

**Comment by fredrikekelund** [comment]
I'm guessing that #152 is a duplicate of this ticket

**Comment by fredrikekelund** [comment]
I just published simplecrawler 0.6.1. If you were experiencing this issue, please give it a try!
